src/chain/shared_state.rs

```diff
@@ -47,17 +47,20 @@ pub struct SharedState {
     /// Our section's key history for Secure Message Delivery
     pub our_history: SectionProofChain,
     /// BLS public keys of other sections
-    pub their_keys: BTreeMap<Prefix<XorName>, TheirKeyInfo>,
+    pub their_keys: BTreeMap<Prefix<XorName>, SectionKeyInfo>,
     /// Other sections' knowledge of us
     pub their_knowledge: BTreeMap<Prefix<XorName>, u64>,
     /// Recent keys removed from their_keys
-    pub their_recent_keys: VecDeque<(Prefix<XorName>, TheirKeyInfo)>,
+    pub their_recent_keys: VecDeque<(Prefix<XorName>, SectionKeyInfo)>,
 }
 
 impl SharedState {
     pub fn new(section_info: SectionInfo) -> Self {
         let pk = BlsPublicKey::from_section_info(&section_info);
-        let our_history = SectionProofChain::from_genesis(pk);
+        let our_history = SectionProofChain::from_genesis(pk.clone());
```

Comment:
don't see other place using this `pk`
is the `clone` really needed?

Reply:
Will fix.